libs/js-sdk/src/client.ts

```diff
@@ -88,26 +91,48 @@ export function createBrowserClient({
     return {
         client: createBrowserFbClient(state),
         get initialised() {
-            return initialisedPromise.completed
+            return (
+                initialisedPromise.completed &&
+                updatedPromise.every((p) => p.completed)
+            )
         },
         waitForInitialised() {
             return new Promise((resolve) => {
                 const interval = setInterval(() => {
-                    if (initialisedPromise.completed) {
+                    if (this.initialised) {
                         clearInterval(interval)
                         resolve(true)
                     }
                 }, 100)
             })
         },
-        updateAudiences(updatedAudiences: string[]) {
+        initialisedChanged(callback) {
+            initialisedCallback = callback
+        },
+        async updateAudiences(updatedAudiences: string[]) {
             debugLog('Updating audiences: %o', {
                 updatedAudiences,
+                initialised: this.initialised,
             })
-            return updateStrategyImplementation.updateAudiences(
-                state,
-                updatedAudiences,
-            )
+
+            const promise = new PromiseCompletionSource<boolean>()
+
+            updatedPromise.push(promise)
```

Comment:
Why do we need a list here and not just overwrite the initial one?

Reply:
React calls update audience twice when running, at least when running in debug. However, features is only fetched if audiences has changed which means the first change triggers a fetch and a race condition might occur. Also we will set promise resolved on a promise that already has been resolved.